src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/KeyInfo.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Xml;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.IdentityModel.Tokens;
+
+namespace CoreWCF.IdentityModel
+{
+    internal class KeyInfo
+    {
+        SecurityTokenSerializer _keyInfoSerializer;
+        SecurityKeyIdentifier _ski;
+        string _retrieval;
+
+        public KeyInfo( SecurityTokenSerializer keyInfoSerializer )
+        {
+            _keyInfoSerializer = keyInfoSerializer;
+            _ski = new SecurityKeyIdentifier();
+        }
+
+        public string RetrievalMethod
+        {
+            get { return _retrieval; }
+        }
+
+        public SecurityKeyIdentifier KeyIdentifier
+        {
+            get { return _ski; }
+            set
+            {
+                if ( value == null )
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull( nameof(value) );
+                }
```

Comment:
Formatting, there's extra spaces around the nameof. Also on the nameof in the next method, and at the end of the arguments to IsStartElement on line 49. Also extra spaces around the while loop expression on line 54. Lots of extra spaces all through the rest of this file so I'm not going to list them all out.